tutorials/02-conda_envs.rst

```diff
@@ -0,0 +1,114 @@
+An Astropy User's Guide to Managing Conda Environments and Jupyter Notebook
+===========================================================================
+
+
+    Help! My code library works in Python 2 and I don't have time to
+    make it work in Python 3
+
+Do not fear! This tutorial will teach you how to use conda environments
+to create a *separate* installation of Python 2.7. With conda
+environments, you can switch between Python 2 and 3 without having to
+worry about version conflicts.
+
+Step 1: Set up a Python 2 environment
+-------------------------------------
+
+-  Create a new ``conda`` environment
+
+   ::
+
+       conda create -n python2 python=2.7 anaconda 
```

Comment:
I wonder if having them install anaconda is overkill. I typically create a new environment with the minimum required packages that I need (e.g. `conda create -n python2 python=2.7 astropy`). On the other hand I'm not the typical user...

Reply:
I also wondered about this, does anaconda needed, or better start with miniconda for all cases? (I don't know as I don't use conda much beyond CI).

Reply:
For the purposes of this tutorial -- creating a separate Python 2 installation -- a new user will probably want *everything* they might possibly use in both installations. Developers, who likely have many environments, seem more likely to need/want a slimmer installation. I'll work on communicating this in the tutorial (i.e., that `astropy` and `miniconda` are options to replace `anaconda` in this line).  

**Can anyone link me to a handy resource guide on the differences between miniconda and anaconda?** (A Quick google search didn't help)

Reply:
Does this help? https://stackoverflow.com/questions/45421163/anaconda-vs-miniconda

Reply:
@eblur -- I don't know of a great online resource, but maybe this helps (and feel free to add it to the tutorial if you want):

## What is the difference between *miniconda* and *anaconda*?

**miniconda** is a fairly small installer that comes with python, conda and little else. One then uses `conda install...` to install any desired conda packages, or one uses `conda create ....` to create an environment with any packages you want.

**anaconda** is really two things in one: a python installer (like *miniconda*) and a conda *metepackage*. A metapackage is a list of packages and version numbers. Anaconda (the company) works hard to make sure that all of the package versions in anaconda (the metapackage) work well with each other. The full list of packages included with anaconda is at https://docs.anaconda.com/anaconda/packages/pkg-docs

## Which one should I choose?

Although it may be tempting to install *anaconda* because it comes with more packages, in the long run *miniconda* is more flexible and easier to maintain. Part of the motivation for this choice is that it encourages you to create *environments* regularly. That is particularly important if you want to have a separate environment for development and for everyday science use.

Reply:
@mwcraig This text is super useful! Where did you get it? I would like to add verbatim, but feel morally obligated to cite the reference.

Reply:
adding a link to Anconda vs Miniconda on the conda readthedocs